### ISSUE TITLE
updating the enum value does not flag the model as dirty

### ISSUE THREAD

**Issue by panthony** [rmachielse/ember-enum#3]
I have a model such as:

```js
import DS from 'ember-data';

export default DS.Model.extend({

  currency: DS.attr('enum', {
    options: [
      'EUR',
      'USD'
    ],
    defaultValue: 'EUR'
  })

});

```

Updating the model's value won't flag it as dirty:

```js
 model.set('currency.value', 'USD');
 console.log(model.get('hasDirtyAttributes')); // false
```

**Comment by rmachielse** [comment]
@panthony yes that's another problem with this approach. Maybe we should change the addon, so the enum object will be different from the actual attribute. That will also make it possible to create a model like this:

```javascript
this.get('store').createRecord('invitation', {
  status: 'accepted'
});
```

**Comment by rmachielse** [comment]
@panthony the source attribute would then just be a number or a string

**Comment by panthony** [comment]
@rmachielse I was starting to dig how ember-data knows that an attribute is dirty in order to understand why it does not see it updated or if the returned Ember Object has something to do.

But right now it's launch time

**Comment by panthony** [comment]
@rmachielse Looks difficult to make it work with the current implementation since we do not have the record instance to manage its state.

Maybe the `enum` attribute should return the value as string, and just make sure the value it serialize/unserialize is part of the `options`.

Maybe add a `mixin` to the addon for the model itself that finds attributes of type `enum` and expose `is*` methods + enum's options ?

**Comment by bryanaka** [comment]
@panthony and @rmachielse I ran into this same issue. How would you feel about switching to an API that was like `model.set('currency', 'USD')` vs. `model.set('currency.value', 'USD')`?

I actually think I can fix this issue in a PR if we make that API change.

**Comment by bryanaka** [comment]
side note: `value` would be read only in this new API and the `currency.isUSD` style API would remain the same.

**Comment by panthony** [comment]
@bryanaka I won't mind an API break. I actually stopped using this addon because I did not want to work around this issue.

**Comment by bryanaka** [comment]
as owner of the repo, would like @rmachielse's feedback too (of course) - but will be opening up a PR shortly with a code spike. Then can work with you two to make some adjustments as needed

**Comment by bryanaka** [comment]
@panthony and @rmachielse see https://github.com/rmachielse/ember-enum/pull/4
